File: cpp/arrays/frequency.cpp
#include <bits/stdc++.h>
using namespace std;

unordered_map<int, int> dict;

void findFrequency(int arr[], int n){
    for(int i=0; i<n; i++)
        dict[arr[i]]++;
}

int frequency1(int arr[], int val, int n){
    findFrequency(arr, n);
    return dict[val];
}

int frequency2(int arr[], int val, int n){
    int count = 0;
    for(int i=0; i<n; i++)
        arr[i]==val ? count++ : NULL;
    return count;
}

int main(){
    int arr[] = {1, 3, 5, 3, 6, 7, 1, 0, 3};
    int n = sizeof(arr)/ sizeof(arr[0]);
    int val, freq;
    cout << "Enter value to count: ";
    cin >> val;

//    freq = frequency1(arr, val, n);
    freq = frequency2(arr, val, n);
    cout << "No. of " << val << "'s in array: " << freq;
}

File: Arrays/frequency.py
dict = {}

def findFrequency(arr):
    global dict
    for key in arr:
        dict[key] = [dict.get(key, 0) + 1]
    
def frequency1(arr, val): 
    findFrequency(arr)   
    return dict[val] if val in dict else 0

def frequency2(arr, val):
    count = 0
    for key in arr:
        if key == val:
            count += 1
    return count

arr = [1, 3, 5, 3, 6, 7, 1, 0, 3]
val = int(input('Enter value to count: '))
# freq = frequency1(arr, val)
freq = frequency2(arr, val)
print("No. of {}'s in array: {}".format(val, freq))